packages/dotcom-server-handlebars/src/helpers/renderJsxComponent.ts

```diff
@@ -0,0 +1,36 @@
+import path from 'path'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+
+export default function renderJsxComponent({ hash }) {
+  let modulePath
+
+  if (hash.hasOwnProperty('package')) {
+    modulePath = path.join(process.cwd(), `node_modules/${hash.package}`)
```

Comment:
Can `require.resolve()` be used?

Attempts to use it produced: `Error: Cannot find module 'node_modules/@financial-times/dotcom-ui-header'`.

Reply:
`require.resolve(hash.package)` should be fine

Reply:
Although... because linked dependencies are not flattened you may need to make the header package a direct dependency of the demo app if you're currently consuming via the layout component.

Reply:
I _think_ the error might be thrown because the node module it is trying to require is a symlink rather than a directory.

![Screenshot 2020-02-05 at 13 24 18](https://user-images.githubusercontent.com/10484515/73846667-cdf6a900-481c-11ea-892b-dbc22d1669c2.png)

[This discussion](https://github.com/nodejs/node/issues/18009) seems to propose adding an argument to `require.resolve()` to recognise and handle dependencies that are symlinks, but the [Node.js docs for that module](https://nodejs.org/api/modules.html#modules_require_resolve_request_options) indicate that no such feature was ever added.

Reply:
Nope, I have disproved this suspicion by re-installing node modules having changed `dependencies` from:
`"@financial-times/dotcom-ui-header": "file:../../packages/dotcom-ui-header"`
to:
`"@financial-times/dotcom-ui-header": "^0.5.8"`
which produces same error.

Reply:
This works:
- ``require(path.resolve(`node_modules/${hash.package}`))``

These do not:
- ``require.resolve(`node_modules/${hash.package}`)``
- `require.resolve(hash.package)`

Will keep reading until I full understand what is going on and why `require.resolve()` is not working as expected.

Reply:
Fixed in [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/746/commits/a79f15f4038f6b153726209e400f500647e0e929).